rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -273,17 +273,18 @@ Node::declare_parameters(
   bool ignore_overrides)
 {
   std::vector<ParameterT> result;
-  std::string normalized_namespace = namespace_.empty() ? "" : (namespace_ + ".");
-  std::transform(
-    parameters.begin(), parameters.end(), std::back_inserter(result),
-    [this, &normalized_namespace, ignore_overrides](auto element) {
-      return this->declare_parameter(
-        normalized_namespace + element.first,
-        element.second,
-        rcl_interfaces::msg::ParameterDescriptor(),
-        ignore_overrides);
-    }
-  );
+  std::vector<
+    std::pair<rclcpp::Parameter, rcl_interfaces::msg::ParameterDescriptor>> parameters_;
+  for (const auto & [name, value] : parameters) {
+    parameters_.push_back(
+      std::make_pair(
+        rclcpp::Parameter(name, value),
+        rcl_interfaces::msg::ParameterDescriptor()));
+  }
+  auto results = this->declare_parameters(namespace_, parameters_, ignore_overrides);
+  for (const auto & res : results) {
+    result.push_back(res.get<ParameterT>());
+  }
   return result;
```

Comment:
What is the purpose of this `for` loop? Both signatures for `declare_parameters` return a `std::vector<rclcpp::ParameterValue>` right?
```suggestion
  return this->declare_parameters(namespace_, parameters_, ignore_overrides);
```

Reply:
Yes. We need make some convert to fit for declare_parameters template.